### PR TITLE
Allow to send Infinite double values either as max or min

### LINKED ISSUE #1264: conversion of score from double to byte array should consider both "-inf" and "+inf"
Below logic is what Jedis converts scores typed double to byte arrays. 

```
    byte[] byteArrayMin = (min == Double.NEGATIVE_INFINITY) ? "-inf".getBytes() : toByteArray(min);
    byte[] byteArrayMax = (max == Double.POSITIVE_INFINITY) ? "+inf".getBytes() : toByteArray(max);
```

I think this kind of conversion is bad cause giving "-inf" "-inf" as score range is still valid. 
Conversion of score should check both Double.NEGATIVE_INFINITY and Double.POSITIVE_INFINITY for each score, not handling by context.

### PR DESCRIPTION
**Pull request by marcosnils**
Fixes #1264 and fixes #1271

### CODE REVIEW

File: src/main/java/redis/clients/jedis/BinaryClient.java [hunk 1]
```diff
@@ -703,10 +685,7 @@ public void zrangeByScoreWithScores(final byte[] key, final String min, final St
 
   public void zrevrangeByScoreWithScores(final byte[] key, final double max, final double min) {
 
-    byte[] byteArrayMin = (min == Double.NEGATIVE_INFINITY) ? "-inf".getBytes() : toByteArray(min);
-    byte[] byteArrayMax = (max == Double.POSITIVE_INFINITY) ? "+inf".getBytes() : toByteArray(max);
-
-    sendCommand(ZREVRANGEBYSCORE, key, byteArrayMax, byteArrayMin, WITHSCORES.raw);
+    sendCommand(ZREVRANGEBYSCORE, key, toByteArray(min), toByteArray(max), WITHSCORES.raw);
```
Comment (HeartSaVioR) [inline src/main/java/redis/clients/jedis/BinaryClient.java]: parameter sequence swapped: max, min is correct.

File: src/main/java/redis/clients/jedis/BinaryClient.java [hunk 2]
```diff
@@ -674,10 +662,7 @@ public void zrangeByScore(final byte[] key, final String min, final String max,
   public void zrevrangeByScore(final byte[] key, final double max, final double min,
       final int offset, int count) {
 
-    byte[] byteArrayMin = (min == Double.NEGATIVE_INFINITY) ? "-inf".getBytes() : toByteArray(min);
-    byte[] byteArrayMax = (max == Double.POSITIVE_INFINITY) ? "+inf".getBytes() : toByteArray(max);
-
-    sendCommand(ZREVRANGEBYSCORE, key, byteArrayMax, byteArrayMin, LIMIT.raw, toByteArray(offset),
+    sendCommand(ZREVRANGEBYSCORE, key, toByteArray(min), toByteArray(max), LIMIT.raw, toByteArray(offset),
```
Comment (HeartSaVioR) [inline src/main/java/redis/clients/jedis/BinaryClient.java]: parameter sequence swapped: max, min is correct.

File: src/main/java/redis/clients/jedis/BinaryClient.java [hunk 3]
```diff
@@ -640,10 +634,7 @@ public void zrangeByScore(final byte[] key, final String min, final String max)
 
   public void zrevrangeByScore(final byte[] key, final double max, final double min) {
 
-    byte[] byteArrayMin = (min == Double.NEGATIVE_INFINITY) ? "-inf".getBytes() : toByteArray(min);
-    byte[] byteArrayMax = (max == Double.POSITIVE_INFINITY) ? "+inf".getBytes() : toByteArray(max);
-
-    sendCommand(ZREVRANGEBYSCORE, key, byteArrayMax, byteArrayMin);
+    sendCommand(ZREVRANGEBYSCORE, key, toByteArray(min), toByteArray(max));
```
Comment (HeartSaVioR) [inline src/main/java/redis/clients/jedis/BinaryClient.java]: parameter sequence swapped: max, min is correct.

File: src/main/java/redis/clients/jedis/BinaryClient.java [hunk 4]
```diff
@@ -732,10 +708,7 @@ public void zrangeByScoreWithScores(final byte[] key, final String min, final St
   public void zrevrangeByScoreWithScores(final byte[] key, final double max, final double min,
       final int offset, final int count) {
 
-    byte[] byteArrayMin = (min == Double.NEGATIVE_INFINITY) ? "-inf".getBytes() : toByteArray(min);
-    byte[] byteArrayMax = (max == Double.POSITIVE_INFINITY) ? "+inf".getBytes() : toByteArray(max);
-
-    sendCommand(ZREVRANGEBYSCORE, key, byteArrayMax, byteArrayMin, LIMIT.raw, toByteArray(offset),
+    sendCommand(ZREVRANGEBYSCORE, key, toByteArray(min), toByteArray(max), LIMIT.raw, toByteArray(offset),
```
Comment (HeartSaVioR) [inline src/main/java/redis/clients/jedis/BinaryClient.java]: parameter sequence swapped: max, min is correct.

File: src/main/java/redis/clients/jedis/Protocol.java [hunk 5]
```diff
@@ -224,6 +224,9 @@ public static Object read(final RedisInputStream is) {
   }
 
   public static final byte[] toByteArray(final double value) {
+    if (Double.isInfinite(value)) {
```
Comment (HeartSaVioR) [inline src/main/java/redis/clients/jedis/Protocol.java]: That's what I wanted! :)

**Comment by marcosnils** [comment]
@HeartSaVioR comments addressed.

**Comment by HeartSaVioR** [comment]
LGTM!

**Comment by HeartSaVioR** [comment]
Merged into master, 2.8, 2.9 respectively. Thanks for the great work!

**Pull request merged**